Update instructions to run Antrea manually

tnqn opened the pull request:
This patch adds example commands to run Antrea components. It also puts
RBAC resources to separate files for ease of creating RBAC resources only.

Fixes #107

tnqn commented:
/skip-e2e

antoninbas commented on docs/manual-installation.md:
```diff
@@ -27,24 +53,51 @@ Open vSwitch >= 2.8.0 userspace daemon `ovs-vswitchd` and `ovsdb-server` should
 clientConnection:
   kubeconfig: <PATH_TO_KUBE_CONF>
+antreaClientConnection:
+  kubeconfig: <PATH_TO_ANTREA_KUBE_CONF>
```
maybe make it clear that `PATH_TO_ANTREA_KUBE_CONF` is the file from the previous step (3)

tnqn replied:
Done

tnqn commented:
> I think this is getting very complicated. If we do expect this to be useful for developers (I would argue that creating a cluster with Kind is probably a better workflow), maybe we should create scripts to start the agent & controller processes, and make sure these scripts do not break by adding a CI job.

It's a **manual** guide to run Antrea. I think most of containerized applications do have these guides (e.g. [flannel](https://github.com/coreos/flannel/blob/master/Documentation/running.md)), otherwise it's difficult for developers who want to run the application as a process when developing a new feature or even with some IDE debug tool.  It has a different purpose from Kind workflow or yaml deployment.

Adding a CI job to ensure it's not broken is a good idea, it's just that I don't think the configuration different from in-cluster mode (which is mainly the guide includes) will change a lot in a short time. I would prefer adding the script and validation when it becomes necessary.

antoninbas commented:
@tnqn I think my point is that having scripts would make that process much easier. The flannel instructions you pointed me to are much more straightforward than what we have now. Is there a fundamental reason why we can't have `./run_controller.sh <some args>` and `./run_agent.sh <some args>`?

tnqn commented:
Are you suggesting to replace this guide with several scripts or add additional scripts? I think this is still needed as a basic guide, I could try adding a wrapper script (with another PR) but I feel it wouldn’t  be much simpler than the doc instructions, as it would end up with some other document about the prerequisites of the script and the usage of the new introduced script arguments. The processes are already simple to run, it just takes one config file, which has some configuration that have to be different from in-cluster mode and the doc is mainly explaining it. TBH I don’t see it’s much complicated than flannel and other tools, not sure if it’s just because I wrote it. Which complicated steps are you referring to or all of them?

antoninbas commented:
@tnqn I think a good compromise if you don't like the script idea is to show some typical commands for steps 1 & 2 of the `antrea-controller` section. Additionally - and maybe I'm missing something here -, isn't `ANTREA_CONTROLLER_ENDPOINT` just going to be `localhost` for 99% of people?

tnqn commented:
@antoninbas I added typical commands as you suggested, extracted rbac resources to separate yaml for ease of applying separately (it's also a typical way yamls are organized, e.g. [kube-proxy-rbac.yaml](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/kube-proxy/kube-proxy-rbac.yaml)). 
I'm perhaps the 1% who will still run antrea-controller as pod and point its service address to antrea-agent when developing antrea-agent separately. Anyway I specified it as "localhost" to save people from must editing the script once.

ksamoray commented:
I would add a script to copy the binaries to /opt/antrea or /usr/local/bin, setup systemd startup scripts for controller, agent and create the conf files.
That would make the manual installation doc very simple and will also make the manual install itself very clear maybe.

tnqn commented:
@ksamoray thanks for your input. There could be many things can be done with scripts to adapt different platform/environment. However, the purpose of the doc is only to provide guide to explain how to do it before automating it with scripts, like what other software would normally have, e.g. [flannel](https://github.com/coreos/flannel/blob/master/Documentation/running.md), [calico](https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/integration).
I'm open to have scripts to it if they are useful to people but I think it's out of scope of this PR, which is fixing something that is broken in current doc.

antoninbas commented on docs/manual-installation.md:
````diff
@@ -6,56 +6,112 @@ There are four components which need to be deployed in order to run Antrea:
+1. Grant the `antrea-controller` ServiceAccount necessary permissions to Kubernetes APIs. You can apply
+[controller-rbac.yaml](/build/yamls/base/controller-rbac.yml) to do it.
+```shell script
+kubectl apply -f https://raw.githubusercontent.com/vmware-tanzu/antrea/master/build/yamls/base/controller-rbac.yml
````
Not 100% sure but should we assume that the target audience has cloned the repo and use the relative path (from the root of the repo) here: `build/yamls/base/controller-rbac.yml`? After all if they have built the binaries manually (agent / controller) they must have cloned the repo. That only really matters if they are making changes to the YAML files though.

tnqn replied:
Just trying to keep the command simply to run with less assumption and prerequisites. 
No strong opinion, I will change if you think local relative path is better.

antoninbas commented on docs/manual-installation.md:
````diff
@@ -6,56 +6,112 @@ There are four components which need to be deployed in order to run Antrea:
+1. Grant the `antrea-agent` ServiceAccount necessary permissions to Kubernetes APIs. You can apply [agent-rbac.yaml](
+/build/yamls/base/agent-rbac.yml) to do it.
+```shell script
+kubectl apply -f https://raw.githubusercontent.com/vmware-tanzu/antrea/master/build/yamls/base/agent-rbac.yml
````
same question as above

ksamoray commented:
IDK if it worths mentioning , antrea-agent requires creation of /var/run/antrea

tnqn commented:
> IDK if it worths mentioning , antrea-agent requires creation of /var/run/antrea

@ksamoray It's definitely worth! thanks for reporting, I created #167 to fix it.

tnqn commented:
/skip-e2e

ksamoray commented on docs/manual-installation.md:
````diff
@@ -6,56 +6,112 @@ There are four components which need to be deployed in order to run Antrea:
+
+4. Start `antrea-controller`.
+```shell script
+antrea-controller --config antrea-controller.conf
````
```suggestion
bin/antrea-controller --config antrea-controller.conf
```

ksamoray commented on docs/manual-installation.md:
````diff
@@ -6,56 +6,112 @@ There are four components which need to be deployed in order to run Antrea:
 5. Start `antrea-agent`.
-```
+```shell script
 antrea-agent --config antrea-agent.conf
````
```suggestion
bin/antrea-agent --config antrea-agent.conf
```

tnqn replied:
Applied suggestions, thanks @ksamoray

tnqn commented:
/skip-e2e

tnqn commented:
@antoninbas do you have new comments? If not, I'm going to merge this.

/retest

tnqn commented:
Squashed manually to trigger the missing "Integration" test

/skip-e2e
